patches/api/0419-Add-event-for-player-editing-sign.patch

```diff
@@ -0,0 +1,80 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: by77er <by77er@gmail.com>
+Date: Sat, 10 Jun 2023 19:06:24 -0400
+Subject: [PATCH] Add event for player editing sign
+
+
+diff --git a/src/main/java/io/papermc/paper/event/player/PlayerEditSignEvent.java b/src/main/java/io/papermc/paper/event/player/PlayerEditSignEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..18b8ff5488f9e81733e2b040d43d9a1630e72d66
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/event/player/PlayerEditSignEvent.java
+@@ -0,0 +1,68 @@
++package io.papermc.paper.event.player;
++
++import org.bukkit.block.Sign;
++import org.bukkit.block.sign.Side;
++import org.bukkit.entity.Player;
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.player.PlayerEvent;
++import org.jetbrains.annotations.NotNull;
++
++/**
++ * Called when a player edits a sign
++ * <p>
++ * Cancelling this event stops the sign editing menu from appearing.
++ */
++public class PlayerEditSignEvent extends PlayerEvent implements Cancellable {
```

Comment:
I wonder if this name properly communicates the fact that this event isn't really called when the edit happens but when the player opens the editing screen.

Maybe it'd be great to reiterate if we can find a more specific name here.

Reply:
Maybe just `PlayerOpenSignEvent`? Least clunky name I could think of, if a little vague.